libarchive/archive_read_support_format_rar5.c

```diff
@@ -1715,9 +1715,19 @@ static int process_head_file(struct archive_read* a, struct rar5* rar,
 	/* If we're currently switching volumes, ignore the new definition of
 	 * window_size. */
 	if(rar->cstate.switch_multivolume == 0) {
-		/* Values up to 64M should fit into ssize_t on every
-		 * architecture. */
-		rar->cstate.window_size = (ssize_t) window_size;
+		if(rar->cstate.window_size != (ssize_t) window_size) {
+			/* Values up to 64M should fit into ssize_t on every
+			 * architecture. */
+			rar->cstate.window_size = (ssize_t) window_size;
+
+			if(rar->cstate.window_buf) {
+				/* If window_buf has been allocated before, reallocate it, so
+				 * that its size will match new window_size. */
+
+				free(rar->cstate.window_buf);
+				rar->cstate.window_buf = calloc(1, rar->cstate.window_size);
```

Comment:
Two problems here. It should be using realloc and also handle a failed allocation.

Reply:
Right. I'll fix it.